### ISSUE TITLE
Fatal Error Composer Install bec. PhpParser?

### ISSUE THREAD

**Issue by tuefekci** [composer-unused/composer-unused#232]
## Describe the bug
Installed via $ composer global require icanhazstring/composer-unused and does not just work out the box living these here because i know that these type of problems are sometimes hard to find.

Happens for me with global and local installation the phar is probably fine if vendors are bundled.

## Error dump

```
PHP Fatal error:  Uncaught Error: Interface 'PhpParser\ErrorHandler' not found in /home/tuefekci/.config/composer/vendor/icanhazstring/composer-unused/src/Error/ErrorHandlerInterface.php:10
Stack trace:
#0 phar:///usr/local/bin/composer/vendor/composer/ClassLoader.php(571): include()
#1 phar:///usr/local/bin/composer/vendor/composer/ClassLoader.php(428): Composer\Autoload\includeFile()
#2 [internal function]: Composer\Autoload\ClassLoader->loadClass()
#3 /home/tuefekci/.config/composer/vendor/icanhazstring/composer-unused/src/Error/ErrorHandler.php(11): spl_autoload_call()
#4 phar:///usr/local/bin/composer/vendor/composer/ClassLoader.php(571): include('/home/tuefekci/...')
#5 phar:///usr/local/bin/composer/vendor/composer/ClassLoader.php(428): Composer\Autoload\includeFile()
#6 [internal function]: Composer\Autoload\ClassLoader->loadClass()
#7 /home/tuefekci/.config/composer/vendor/icanhazstring/composer-unused/src/Error/Factory/ErrorHandlerFactory.php(16): spl_autoload_call()
#8 /home/tuefekci/.config/composer/vendor/icanhazs in /home/tuefekci/.config/composer/vendor/icanhazstring/composer-unused/src/Error/ErrorHandlerInterface.php on line 10

Fatal error: Uncaught Error: Interface 'PhpParser\ErrorHandler' not found in /home/tuefekci/.config/composer/vendor/icanhazstring/composer-unused/src/Error/ErrorHandlerInterface.php:10
Stack trace:
#0 phar:///usr/local/bin/composer/vendor/composer/ClassLoader.php(571): include()
#1 phar:///usr/local/bin/composer/vendor/composer/ClassLoader.php(428): Composer\Autoload\includeFile()
#2 [internal function]: Composer\Autoload\ClassLoader->loadClass()
#3 /home/tuefekci/.config/composer/vendor/icanhazstring/composer-unused/src/Error/ErrorHandler.php(11): spl_autoload_call()
#4 phar:///usr/local/bin/composer/vendor/composer/ClassLoader.php(571): include('/home/tuefekci/...')
#5 phar:///usr/local/bin/composer/vendor/composer/ClassLoader.php(428): Composer\Autoload\includeFile()
#6 [internal function]: Composer\Autoload\ClassLoader->loadClass()
#7 /home/tuefekci/.config/composer/vendor/icanhazstring/composer-unused/src/Error/Factory/ErrorHandlerFactory.php(16): spl_autoload_call()
#8 /home/tuefekci/.config/composer/vendor/icanhazs in /home/tuefekci/.config/composer/vendor/icanhazstring/composer-unused/src/Error/ErrorHandlerInterface.php on line 10
```

## Additional information
```
  Problem 1
    - Root composer.json requires icanhazstring/composer-unused ^0.7.7 -> satisfiable by icanhazstring/composer-unused[0.7.7].
    - icanhazstring/composer-unused 0.7.7 requires nikic/php-parser ^4.12 -> found nikic/php-parser[v4.12.0, v4.13.0] but the package is fixed to v1.4.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
```

**Comment by icanhazstring** [comment]
Hi @tuefekci, a global require is more problematic as a local one.

If you require it globally, your local requirements will be loaded instead of global ones. This will probably cause the ClassNotFound and the composer req error since you are using php-parser 1.4, while I require 4.12.

To maybe fix this, you can try the following:

```
composer rem nikic/php-parser
composer icanhazstring/composer-unused -W
```

Beware that this will upgrade all 3rd party dependencies composer-unused is using that are installed on your project.

**Comment by tuefekci** [comment]
```
~ composer rem nikic/php-parser                                                                                                               2 [03:12:33]
nikic/php-parser is not required in your composer.json and has not been removed
```

It's probably a requirement of another package? It's not a big deal for me, I only opened the issue so less experienced people etc. will not fall into ii being unusable, So using the phar is really the optimal way like you write in the README.

I just read that the local part 
`exclamation Beware: Local (or global) requirement might lead to issues related to outdated or replaced dependencies and composer-unused might not work as intended!`

Perhaps adding it differently in the README would be better because to tell the truth I did not read the local part at all because that was not what I was trying to install.
